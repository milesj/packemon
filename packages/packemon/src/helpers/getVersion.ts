import fs from 'fs';
import { PackageStructure } from '@boost/common';

export function getVersion(): string {
	return (
		JSON.parse(
			// @ts-expect-error Not typed for URL
			fs.readFileSync(new URL('../../package.json', import.meta.url), 'utf8'),
		) as PackageStructure
	).version;
}
