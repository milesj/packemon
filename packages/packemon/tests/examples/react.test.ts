import { testExampleOutput } from '../helpers';

// This is skipped since Rollup seems to crash with a syntax error,
// and I've been unable to figure out why...
// eslint-disable-next-line jest/no-disabled-tests
describe.skip('React/JSX', () => {
	const oldEnv = process.env.NODE_ENV;

	beforeEach(() => {
		// Development includes host machine file paths
		process.env.NODE_ENV = 'production';
	});

	afterEach(() => {
		process.env.NODE_ENV = oldEnv;
	});

	// Automatic only
	testExampleOutput('react.tsx', 'babel');
});
