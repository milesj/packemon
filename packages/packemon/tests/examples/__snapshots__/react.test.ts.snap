// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React/JSX babel transforms example test case: browser-current-esm 1`] = `
[
  "esm/index-browser-current-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: esm

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-current-esm.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-current-lib 1`] = `
[
  "lib/index-browser-current-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-current-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-current-umd 1`] = `
[
  "umd/index-browser-current-umd.js",
  "(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react/jsx-runtime"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react/jsx-runtime"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.jsxRuntime);
    global.examples = mod.exports;
  }
})(typeof globalThis !== "undefined" ? globalThis : typeof self !== "undefined" ? self : this, function (_exports, _jsxRuntime) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.default = Button;
  // Bundled with Packemon: https://packemon.dev
  // Platform: browser, Support: current, Format: umd

  function Button() {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("button", {});
  }
});
//# sourceMappingURL=index-browser-current-umd.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-experimental-esm 1`] = `
[
  "esm/index-browser-experimental-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: esm

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-experimental-esm.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-experimental-lib 1`] = `
[
  "lib/index-browser-experimental-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-experimental-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-experimental-umd 1`] = `
[
  "umd/index-browser-experimental-umd.js",
  "(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react/jsx-runtime"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react/jsx-runtime"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.jsxRuntime);
    global.examples = mod.exports;
  }
})(typeof globalThis !== "undefined" ? globalThis : typeof self !== "undefined" ? self : this, function (_exports, _jsxRuntime) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.default = Button;
  // Bundled with Packemon: https://packemon.dev
  // Platform: browser, Support: experimental, Format: umd

  function Button() {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("button", {});
  }
});
//# sourceMappingURL=index-browser-experimental-umd.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-legacy-esm 1`] = `
[
  "esm/index-browser-legacy-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: esm

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-legacy-esm.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-legacy-lib 1`] = `
[
  "lib/index-browser-legacy-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: lib

'use strict';

var jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-legacy-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-legacy-umd 1`] = `
[
  "umd/index-browser-legacy-umd.js",
  "(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react/jsx-runtime"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react/jsx-runtime"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.jsxRuntime);
    global.examples = mod.exports;
  }
})(typeof globalThis !== "undefined" ? globalThis : typeof self !== "undefined" ? self : this, function (_exports, _jsxRuntime) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.default = Button;
  // Bundled with Packemon: https://packemon.dev
  // Platform: browser, Support: legacy, Format: umd

  function Button() {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("button", {});
  }
});
//# sourceMappingURL=index-browser-legacy-umd.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-stable-esm 1`] = `
[
  "esm/index-browser-stable-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: esm

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-stable-esm.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-stable-lib 1`] = `
[
  "lib/index-browser-stable-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-stable-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: browser-stable-umd 1`] = `
[
  "umd/index-browser-stable-umd.js",
  "(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react/jsx-runtime"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react/jsx-runtime"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.jsxRuntime);
    global.examples = mod.exports;
  }
})(typeof globalThis !== "undefined" ? globalThis : typeof self !== "undefined" ? self : this, function (_exports, _jsxRuntime) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.default = Button;
  // Bundled with Packemon: https://packemon.dev
  // Platform: browser, Support: stable, Format: umd

  function Button() {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("button", {});
  }
});
//# sourceMappingURL=index-browser-stable-umd.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: native-current-lib 1`] = `
[
  "lib/index-native-current-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: current, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-current-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: native-experimental-lib 1`] = `
[
  "lib/index-native-experimental-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: experimental, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-experimental-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: native-legacy-lib 1`] = `
[
  "lib/index-native-legacy-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: legacy, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-legacy-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: native-stable-lib 1`] = `
[
  "lib/index-native-stable-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: stable, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-stable-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-current-cjs 1`] = `
[
  "cjs/index-node-current-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-current-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX babel transforms example test case: node-current-cjs 2`] = `
[
  "cjs/index-node-current-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: cjs

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-current-cjs.cjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-current-lib 1`] = `
[
  "lib/index-node-current-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-current-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-current-mjs 1`] = `
[
  "mjs/index-node-current-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: mjs

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-current-mjs.mjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-experimental-cjs 1`] = `
[
  "cjs/index-node-experimental-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-experimental-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX babel transforms example test case: node-experimental-cjs 2`] = `
[
  "cjs/index-node-experimental-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: cjs

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-experimental-cjs.cjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-experimental-lib 1`] = `
[
  "lib/index-node-experimental-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-experimental-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-experimental-mjs 1`] = `
[
  "mjs/index-node-experimental-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: mjs

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-experimental-mjs.mjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-legacy-cjs 1`] = `
[
  "cjs/index-node-legacy-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-legacy-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX babel transforms example test case: node-legacy-cjs 2`] = `
[
  "cjs/index-node-legacy-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: cjs

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-legacy-cjs.cjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-legacy-lib 1`] = `
[
  "lib/index-node-legacy-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-legacy-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-legacy-mjs 1`] = `
[
  "mjs/index-node-legacy-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: mjs

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-legacy-mjs.mjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-stable-cjs 1`] = `
[
  "cjs/index-node-stable-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-stable-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX babel transforms example test case: node-stable-cjs 2`] = `
[
  "cjs/index-node-stable-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: cjs

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-stable-cjs.cjs.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-stable-lib 1`] = `
[
  "lib/index-node-stable-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: lib

'use strict';

const jsxRuntime = require('react/jsx-runtime');
function Button() {
  return /*#__PURE__*/jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-stable-lib.js.map
",
]
`;

exports[`React/JSX babel transforms example test case: node-stable-mjs 1`] = `
[
  "mjs/index-node-stable-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: mjs

import { jsx } from 'react/jsx-runtime';
function Button() {
  return /*#__PURE__*/jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-stable-mjs.mjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-current-esm 1`] = `
[
  "esm/index-browser-current-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: esm
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-current-esm.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-current-lib 1`] = `
[
  "lib/index-browser-current-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-current-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-current-umd 1`] = `
[
  "umd/index-browser-current-umd.js",
  "(function(global, factory) {
    if (typeof module === "object" && typeof module.exports === "object") factory(exports, require("react/jsx-runtime"));
    else if (typeof define === "function" && define.amd) define([
        "exports",
        "react/jsx-runtime"
    ], factory);
    else if (global = typeof globalThis !== "undefined" ? globalThis : global || self) factory(global.examples = {}, global.jsxRuntime);
})(this, function(exports, _jsxRuntime) {
    "use strict";
    Object.defineProperty(exports, "__esModule", {
        value: true
    });
    Object.defineProperty(exports, "default", {
        enumerable: true,
        get: ()=>Button
    });
    function Button() {
        return /*#__PURE__*/ (0, _jsxRuntime.jsx)("button", {});
    }
});
//# sourceMappingURL=index-browser-current-umd.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-experimental-esm 1`] = `
[
  "esm/index-browser-experimental-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: esm
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-experimental-esm.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-experimental-lib 1`] = `
[
  "lib/index-browser-experimental-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-experimental-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-experimental-umd 1`] = `
[
  "umd/index-browser-experimental-umd.js",
  "(function(global, factory) {
    if (typeof module === "object" && typeof module.exports === "object") factory(exports, require("react/jsx-runtime"));
    else if (typeof define === "function" && define.amd) define([
        "exports",
        "react/jsx-runtime"
    ], factory);
    else if (global = typeof globalThis !== "undefined" ? globalThis : global || self) factory(global.examples = {}, global.jsxRuntime);
})(this, function(exports, _jsxRuntime) {
    "use strict";
    Object.defineProperty(exports, "__esModule", {
        value: true
    });
    Object.defineProperty(exports, "default", {
        enumerable: true,
        get: ()=>Button
    });
    function Button() {
        return /*#__PURE__*/ (0, _jsxRuntime.jsx)("button", {});
    }
});
//# sourceMappingURL=index-browser-experimental-umd.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-legacy-esm 1`] = `
[
  "esm/index-browser-legacy-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: esm
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-legacy-esm.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-legacy-lib 1`] = `
[
  "lib/index-browser-legacy-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: lib
'use strict';
var jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-legacy-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-legacy-umd 1`] = `
[
  "umd/index-browser-legacy-umd.js",
  "(function(global, factory) {
    if (typeof module === "object" && typeof module.exports === "object") factory(exports, require("react/jsx-runtime"));
    else if (typeof define === "function" && define.amd) define([
        "exports",
        "react/jsx-runtime"
    ], factory);
    else if (global = typeof globalThis !== "undefined" ? globalThis : global || self) factory(global.examples = {}, global.jsxRuntime);
})(this, function(exports, _jsxRuntime) {
    "use strict";
    Object.defineProperty(exports, "__esModule", {
        value: true
    });
    Object.defineProperty(exports, "default", {
        enumerable: true,
        get: function() {
            return Button;
        }
    });
    function Button() {
        return /*#__PURE__*/ (0, _jsxRuntime.jsx)("button", {});
    }
});
//# sourceMappingURL=index-browser-legacy-umd.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-stable-esm 1`] = `
[
  "esm/index-browser-stable-esm.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: esm
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-browser-stable-esm.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-stable-lib 1`] = `
[
  "lib/index-browser-stable-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-browser-stable-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: browser-stable-umd 1`] = `
[
  "umd/index-browser-stable-umd.js",
  "(function(global, factory) {
    if (typeof module === "object" && typeof module.exports === "object") factory(exports, require("react/jsx-runtime"));
    else if (typeof define === "function" && define.amd) define([
        "exports",
        "react/jsx-runtime"
    ], factory);
    else if (global = typeof globalThis !== "undefined" ? globalThis : global || self) factory(global.examples = {}, global.jsxRuntime);
})(this, function(exports, _jsxRuntime) {
    "use strict";
    Object.defineProperty(exports, "__esModule", {
        value: true
    });
    Object.defineProperty(exports, "default", {
        enumerable: true,
        get: ()=>Button
    });
    function Button() {
        return /*#__PURE__*/ (0, _jsxRuntime.jsx)("button", {});
    }
});
//# sourceMappingURL=index-browser-stable-umd.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: native-current-lib 1`] = `
[
  "lib/index-native-current-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: current, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-current-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: native-experimental-lib 1`] = `
[
  "lib/index-native-experimental-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: experimental, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-experimental-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: native-legacy-lib 1`] = `
[
  "lib/index-native-legacy-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: legacy, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-legacy-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: native-stable-lib 1`] = `
[
  "lib/index-native-stable-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: native, Support: stable, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-native-stable-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-current-cjs 1`] = `
[
  "cjs/index-node-current-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-current-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX swc transforms example test case: node-current-cjs 2`] = `
[
  "cjs/index-node-current-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: cjs
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-current-cjs.cjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-current-lib 1`] = `
[
  "lib/index-node-current-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-current-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-current-mjs 1`] = `
[
  "mjs/index-node-current-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: mjs
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-current-mjs.mjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-experimental-cjs 1`] = `
[
  "cjs/index-node-experimental-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-experimental-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX swc transforms example test case: node-experimental-cjs 2`] = `
[
  "cjs/index-node-experimental-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: cjs
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-experimental-cjs.cjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-experimental-lib 1`] = `
[
  "lib/index-node-experimental-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-experimental-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-experimental-mjs 1`] = `
[
  "mjs/index-node-experimental-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: mjs
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-experimental-mjs.mjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-legacy-cjs 1`] = `
[
  "cjs/index-node-legacy-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-legacy-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX swc transforms example test case: node-legacy-cjs 2`] = `
[
  "cjs/index-node-legacy-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: cjs
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-legacy-cjs.cjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-legacy-lib 1`] = `
[
  "lib/index-node-legacy-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-legacy-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-legacy-mjs 1`] = `
[
  "mjs/index-node-legacy-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: mjs
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-legacy-mjs.mjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-stable-cjs 1`] = `
[
  "cjs/index-node-stable-cjs-wrapper.mjs",
  "// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-stable-cjs.cjs';

export default data;",
]
`;

exports[`React/JSX swc transforms example test case: node-stable-cjs 2`] = `
[
  "cjs/index-node-stable-cjs.cjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: cjs
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-stable-cjs.cjs.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-stable-lib 1`] = `
[
  "lib/index-node-stable-lib.js",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: lib
'use strict';
const jsxRuntime = require('react/jsx-runtime');
function Button() {
    return /*#__PURE__*/ jsxRuntime.jsx("button", {});
}
module.exports = Button;
//# sourceMappingURL=index-node-stable-lib.js.map
",
]
`;

exports[`React/JSX swc transforms example test case: node-stable-mjs 1`] = `
[
  "mjs/index-node-stable-mjs.mjs",
  "// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: mjs
import { jsx } from 'react/jsx-runtime';
function Button() {
    return /*#__PURE__*/ jsx("button", {});
}
export { Button as default };
//# sourceMappingURL=index-node-stable-mjs.mjs.map
",
]
`;
